### PR TITLE
[For ruby_3_2] Fix unused_mut Rust warnings

### PR DESCRIPTION
**Pull request by XrXr**
*For YJIT members: please don't merge and let the branch maintainer decide what to do.*

Rust version 1.71.0 and up issue these warnings. On GitHub CI, the
warnings were previously seen in -DYJIT_FORCE_ENABLE runs.

Not a backport, and these code don't exist on the master branch anymore.
Discovered thanks to @nirvdrum.

### CODE REVIEW

**Review by maximecb — APPROVED** [review]
Waiting for branch maintainer to merge as requested.

**Review by nagachika — APPROVED** [review]

**Pull request merged**
